R/remove_funs.R

```diff
@@ -22,12 +22,22 @@ crop <- function ( x , char = " " ) {
 
 ### gsub for more than one pattern
 gsubAll <- function(string, old, new) {
-  stopifnot ( is.character(string), is.character(old), is.character(new), length(old) == length(new))
+  checkmate::assert_character(string)
+  lapply(list(old, new), FUN = checkmate::assert_character,unique=TRUE, any.missing = FALSE, min.len = 2)
+  stopifnot ( length(old) == length(new))
+  stopifnot (length(intersect(old, new))==0)                                    ### in einem Zwischenschritt unique IDs mit gleich vielen characters erzeugen
+  inter <- stringi::stri_rand_strings(n=length(old), length=12, pattern="[A-Za-z0-9]")
+  stopifnot(length(unique(inter)) == length(inter))
+  ind <- sort(nchar(old), decreasing = TRUE, index.return=TRUE)                 ### sortieren nach nchar(), wenn kuerzere strings in laengeren enthalten
+  old <- old[ind[["ix"]]]
+  new <- new[ind[["ix"]]]
+  for(i in seq_along(old)) {
+    string <- gsub(old[i], inter[i], string)
+  }
   for(i in seq_along(old)) {
-    string <- gsub(old[i], new[i], string)
+    string <- gsub(inter[i], new[i], string)
```

Comment:
Very interesting work around!

Reply:
It's a sad, sad story. The problem becomes prevalent in some very rare cases in `eatModel`, when you run DIF analyses in Conquest (because of persistent convergence issues with DIF analyses in TAM). If some item IDs have 11 characters (the maximum Conquest allows), the `item*[dif variable]` statement in the showfile looks odd:
```
TERM 3: item*sex
--------------------------------------------------------------------------------------------------------
          VARIABLES                                       UNWEIGHTED FIT             WEIGHTED FIT
------------------------------                        -----------------------   -----------------------
     item           sex            ESTIMATE  ERROR^   MNSQ       CI        T    MNSQ       CI        T
--------------------------------------------------------------------------------------------------------
 1   M1PTI1     1   0                0.156   0.032    1.05 ( 0.95, 1.05)  2.2   1.02 ( 0.96, 1.04)  0.8 
 2   M1PTI2     1   0               -0.152   0.032    0.81 ( 0.95, 1.05) -8.2   0.92 ( 0.95, 1.05) -3.4 
 3   M1PTI3     1   0                0.146   0.037    0.93 ( 0.95, 1.05) -2.7   0.98 ( 0.94, 1.06) -0.8 
 4   M1PTI6     1   0                0.135   0.028    0.89 ( 0.95, 1.05) -4.5   0.92 ( 0.97, 1.03) -5.5 
 5   M1PT17DIF101   0               -0.217   0.036    0.81 ( 0.95, 1.05) -8.4   0.87 ( 0.95, 1.05) -4.8 
 6   M1PTI11    1   0                0.120   0.033    0.89 ( 0.95, 1.05) -4.8   0.94 ( 0.95, 1.05) -2.6 
 7   M1PTI12    1   0                0.021   0.029    1.21 ( 0.95, 1.05)  8.2   1.11 ( 0.97, 1.03)  6.2
``` 
There is no space sign between variable identifier `M1PT17DIF10` and the numeric DIF group identifier. Reading this ASCII file into R causes problems if column separators are identified by a space character. Reading the ASCII file via `readLines()` yield only character lines that have to be converted into a data.frame. My emergency solution was to insert an additional space between the item column and the DIF identifier column, but at what position exactly? Because the position varies depending on whether there is a two-, three- or four-digit number of items. Conquest reserves two, three or four spaces at the beginning to number the items consecutively. So I used the item identifiers and wanted to replace each individual item ID with the ID itself plus an additional space. And this causes problems when shorter item names are contained in longer ones, i.e. if there is an item `M12` and an item `M123`. 
Welcome to the world of pain!

Reply:
maybe use `gsub(..., fixed = TRUE)` instead?